group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -669,7 +668,12 @@ private void throwIfConsumerGroupHeartbeatRequestIsInvalid(
                 throw new InvalidRequestException("TopicPartitions must be empty when (re-)joining.");
             }
             if (request.subscribedTopicNames() == null || request.subscribedTopicNames().isEmpty()) {
-                throw new InvalidRequestException("SubscribedTopicNames must be set in first request.");
+                if (request.subscribedTopicRegex() == null || request.subscribedTopicRegex().isEmpty()) {
```

Comment:
I would recommend creating 2 separate methods on the lines of => `isSubscribedTopicRegexEmpty()` and `isSubscribedTopicNamesEmpty()` or something similar and moving these if conditions outside. We can use it in the next line as well where we are check if only one of them is set.